SF iOS/SF iOS/Models/EventDataSource.m

```diff
@@ -113,6 +166,19 @@ - (Event *)eventAtIndex:(NSUInteger)index {
     return self.events[index];
 }
 
+/// Updates EventDataSoruce [Event] by text search or gets all events if there is no search term
+///
+/// - paramaters:
+///     -searchTerm: string to search
+/// - returns:
+///     - RLMResults<Event *> *: array of matching Events
+- (RLMResults<Event *> *)filterEventsWithSearchTerm:(NSString *)searchTerm {    
+    NSPredicate *coffeeFilter = [NSPredicate predicateWithFormat:@"name CONTAINS[c] %@", searchTerm];
+    RLMResults<Event *> *filteredCoffee = [[Event objectsWithPredicate:coffeeFilter]
+                                           sortedResultsUsingKeyPath:@"date" ascending:false];
+    return filteredCoffee;
```

Comment:
I do like this a bit better. I wonder if there is a scenario where we change the data source from the controller's perspective so that this DataSource stays pure and simple.

Reply:
Extending on this, I wonder if it's possible to ALSO query on items who's venue contains this. Right now we're only querying on event name. So "Mergesort is in town" wouldn't show up if you searched for "Saint Frank". Could be neat. Actually let me try that.

Reply:
Holy shit!!!
`    NSPredicate *coffeeFilter = [NSPredicate predicateWithFormat:@"name CONTAINS[c] %@ OR venue.name CONTAINS[c] %@ ", searchTerm, searchTerm];`

![laughing](https://media.giphy.com/media/YirLZBJoar89G/giphy.gif)

@mcritz there are a few other changes that need to be made in this PR, can you also do this?